src/utils.ts

```diff
@@ -149,6 +154,135 @@ export function extensionConfigurationSet<
   return getConfiguration().update(key, value, target);
 }
 
+export function canDownloadLanguageServer(): boolean {
+  const platform = os.platform()
+  if (platform != "win32" && platform != "darwin") {
+    return false
+  }
+  const arch = os.arch()
+  return arch == "x64" && findLanguageServer() == null;
+}
+
+function createDownloadURL(): string {
+  const platform = os.platform()
```

Comment:
I think some of these functions should be getters that are only computed the first time. Perhaps creating a `MesonLanguageSever` abstract class, and then you create an implementation for your client. Your client would obviously be the only implementation.

```typescript
abstract class MesonLanguageServer {
  name: string;
  serverOptions: ServerOptions;

  constructor() {
    // This would be your current setupLanguageServer maybe?
  }

  get downloadInfo(): [url: string, hash: string] | null {
    return null;
  }

  private dispose(): void {

  }

  restart(): void {
    // ...
  }
}
```

Hopefully you get what I am saying